### DISCUSSION TITLE
Support for `menuinst`

### DISCUSSION THREAD

**Discussion by dhirschfeld** [prefix-dev/pixi#1894] [Ideas]
### Problem description

It would be great if `pixi` supported `menuinst`:

> This package provides cross platform menu item installation for conda packages.
> 
> If a conda package ships a [menuinst JSON document](https://conda.github.io/menuinst/reference/) under `$PREFIX/Menu`, conda will invoke menuinst to process the JSON file and install the menu items in your operating system. The menu items are removed when the package is uninstalled.

* https://github.com/conda/menuinst

**Comment by tdejager** [comment]
Hey Dave, thanks for the issue 👍
I was thinking about this as well as I currently still don't global install everything through pixi because of missing .app in the osx application folder.


I think this would be great for global installs at least. Do you have any experience how well this works @dhirschfeld ?

**Comment by dhirschfeld** [comment]
I don't have much experience with it, but my use case is also for globally installed tools. 

I don't know that it makes much sense for per-environment tools/libraries but for globally installed tools/applications it could be pretty slick (make for a nice demo!)

**Comment by tdejager** [comment]
I agree, for sure!

**Comment by baszalmstra** [comment]
@tdejager do you think we'd have to port menuinst to rust?

**Comment by tdejager** [comment]
I think so yes, because it would affect the installation procedure. However, we could experiment with the python one first of course.

**Comment by jaimergp** [comment]
Feel free to ping me if there are any questions :) I'm sure we will find some rough edges and opportunities for improvement in `menuinst` itself. The schema should be pretty solid by now, although we are discussing some minor changes now at https://github.com/conda/menuinst/pull/180.

**Comment by baszalmstra** [comment]
I moved this to a discussion in the ideas section. I think its a good idea but I just don't know when we'll get to it.

**Comment by ReimarBauer** [comment]
Without menuinst we found some encoding bugs on windows, e.g. https://github.com/Open-MSS/MSS/issues/2668
menuinst had the great feature not to use a windows terminal and can't run in cp1252 encoding trouble.

**Comment by wolfv** [comment]
Good news - menuinst is coming to `pixi global` very soon! :)

**Reply by phreed** [reply]
Could you provide a link to the relevant, commit, issue, doc, or CEP ...

**Reply by wolfv** [reply]
Most of the work was done in `rattler`, in the `rattler_menuinst` crate: https://github.com/conda/rattler/tree/main/crates/rattler_menuinst

The CEP for menuinst itself is referenced in the menuinst docs: https://conda.github.io/menuinst/

The enablement in `pixi` came here: https://github.com/prefix-dev/pixi/pull/3226

Sorry for not being more diligent with this!

**Reply by phreed** [reply]
CEP: https://github.com/conda/ceps/blob/main/cep-0011.md
rattler-build: https://rattler.build/latest/system_integration/

**Comment by phreed** [comment]
Is this all that is needed to make conda-forge suitable for GUI applications?

I presume some work would be needed to enable this capability in the following applications:?

FreeCAD : https://github.com/conda-forge/freecad-feedstock
Firefox : https://github.com/conda-forge/firefox-feedstock
Gazebo : https://prefix.dev/channels/conda-forge/packages/gazebo
